openstack_controller/datadog_checks/openstack_controller/openstack_controller.py

```diff
@@ -714,47 +661,46 @@ def check(self, instance):
                                                project_scope.nova_endpoint,
                                                project_scope.auth_token)
 
-                project = self.get_and_update_project_details(project_scope)
-                if project and project.get('name'):
-                    projects[project.get('name')] = project
-
-                if collect_limits_from_all_projects:
-                    scope_projects = self.get_projects(project_scope.auth_token)
-                    if scope_projects:
-                        for proj in scope_projects:
-                            projects[proj['name']] = proj
-
-                filtered_projects = pattern_filter([p for p in projects],
-                                                   whitelist=self.include_project_name_rules,
-                                                   blacklist=self.exclude_project_name_rules)
+                self._send_api_service_checks(project_scope, custom_tags)
 
-                projects = {name: v for (name, v) in iteritems(projects) if name in filtered_projects}
+                # List projects and filter them
+                # TODO: NOTE: During authentication we use /v3/auth/projects and here we use /v3/projects.
+                # TODO: These api don't seems to return the same thing however the latter contains the former.
+                # TODO: Is this expected or could we just have one call with proper config?
+                projects = self.get_projects(project_scope.auth_token,
+                                             self.include_project_name_rules,
+                                             self.exclude_project_name_rules)
 
-                for name, project in iteritems(projects):
-                    self.get_stats_for_single_project(project, custom_tags)
+                if collect_project_metrics:
+                    for name, project in iteritems(projects):
+                        self.collect_project_limit(project, custom_tags)
 
-                self.get_stats_for_all_hypervisors(instance, custom_tags=custom_tags,
-                                                   use_shortname=use_shortname,
-                                                   collect_hypervisor_load=collect_hypervisor_load)
+                self.collect_hypervisors_metrics(custom_tags=custom_tags,
+                                                 use_shortname=use_shortname,
+                                                 collect_hypervisor_metrics=collect_hypervisor_metrics,
+                                                 collect_hypervisor_load=collect_hypervisor_load)
 
-                # This updates the server cache directly
-                self.get_all_servers(project_scope.auth_token, instance_name)
-                self.filter_excluded_servers()
```

Comment:
Why remove the ability to filter what servers to collect metrics from based on projects and server ids?

Reply:
It's not removed, I first filter projects and then only fetch servers for those.